testsuite/cluster-test/src/cluster_builder.rs

```diff
@@ -272,11 +323,174 @@ impl ClusterBuilder {
+        let layout = Layout {
+            owners: owners.clone(),
+            operators: owners,
+            libra_root: vec![LIBRA_ROOT_NAME.to_string()],
+        };
+        let layout_path = "/tmp/layout.yaml";
+        write!(File::create(layout_path)?, "{}", toml::to_string(&layout)?)?;
+        let token_path = "/tmp/token";
+        write!(File::create(token_path)?, "{}", VAULT_TOKEN)?;
+        genesis_helper.set_layout(layout_path, "common").await?;
+        genesis_helper
+            .libra_root_key(
+                VAULT_BACKEND,
+                format!("http://{}:{}", vault_nodes[0].internal_ip, VAULT_PORT).as_str(),
+                token_path,
+                LIBRA_ROOT_NAME,
+                LIBRA_ROOT_NAME,
+            )
+            .await?;
+
+        for (i, node) in vault_nodes.iter().enumerate() {
+            genesis_helper
+                .owner_key(
+                    VAULT_BACKEND,
+                    format!("http://{}:{}", node.internal_ip, VAULT_PORT).as_str(),
+                    token_path,
+                    validator_pod_name(i as u32).as_str(),
+                    validator_pod_name(i as u32).as_str(),
+                )
+                .await?;
+            genesis_helper
+                .operator_key(
+                    VAULT_BACKEND,
+                    format!("http://{}:{}", node.internal_ip, VAULT_PORT).as_str(),
+                    token_path,
+                    validator_pod_name(i as u32).as_str(),
+                    validator_pod_name(i as u32).as_str(),
+                )
+                .await?;
+            genesis_helper
+                .validator_config(
+                    validator_pod_name(i as u32).as_str(),
+                    NetworkAddress::from_str(
+                        format!("/ip4/{}/tcp/{}", validator_nodes[i].internal_ip, 6180).as_str(),
+                    )
+                    .expect("Failed to parse network address"),
+                    NetworkAddress::from_str(
+                        format!("/ip4/{}/tcp/{}", fullnode_nodes[i].internal_ip, 6180).as_str(),
+                    )
+                    .expect("Failed to parse network address"),
+                    ChainId::new(1),
+                    VAULT_BACKEND,
+                    format!("http://{}:{}", node.internal_ip, VAULT_PORT).as_str(),
+                    token_path,
+                    validator_pod_name(i as u32).as_str(),
+                    validator_pod_name(i as u32).as_str(),
+                )
+                .await?;
+            genesis_helper
+                .set_operator(
+                    validator_pod_name(i as u32).as_str(),
+                    validator_pod_name(i as u32).as_str(),
+                )
+                .await?;
+        }
+        genesis_helper
+            .genesis(ChainId::new(1), Path::new(GENESIS_PATH))
+            .await?;
+        for (i, node) in vault_nodes.iter().enumerate() {
```

Comment:
do we iterate in same loop on purpose?

Reply:
Yes, genesis_help.genesis() should be called in between these two loops.